### PR TITLE
Run Fuchsia apps without flx

### PR DESCRIPTION
**Pull request by szakarias**
New version of #4538 including review comments and fixes for Windows build problem.

### CODE REVIEW

**Comment by szakarias** [comment]
cc @abarth @jakobr-google

File: assets/directory_asset_bundle.cc [hunk 1]
```diff
@@ -31,7 +58,10 @@ bool DirectoryAssetBundle::GetAsBuffer(const std::string& asset_name,
 DirectoryAssetBundle::~DirectoryAssetBundle() {}
 
 DirectoryAssetBundle::DirectoryAssetBundle(std::string directory)
-    : directory_(std::move(directory)) {}
+    : directory_(std::move(directory)), fd_() {}
+
+DirectoryAssetBundle::DirectoryAssetBundle(fxl::UniqueFD fd)
+    : directory_(""), fd_(std::move(fd)) {}
```
Comment (abarth) [inline assets/directory_asset_bundle.cc]: No need for ("").  The default constructor for std::string makes an empty string.

File: content_handler/runtime_holder.cc [hunk 2]
```diff
@@ -448,9 +454,31 @@ void RuntimeHolder::InitMozartInternal() {
...
 void RuntimeHolder::InitRootBundle(std::vector<char> bundle) {
-  root_bundle_data_ = std::move(bundle);
-  asset_store_ = fxl::MakeRefCounted<blink::ZipAssetStore>(
-      GetUnzipperProviderForRootBundle());
+  if (!bundle.empty()) {
+    root_bundle_data_ = std::move(bundle);
+    asset_store_ = fxl::MakeRefCounted<blink::ZipAssetStore>(
+        GetUnzipperProviderForRootBundle());
+  } else {
+#if defined(OS_WIN)
```
Comment (abarth) [inline content_handler/runtime_holder.cc]: This file isn't compiled on Windows, so you can remove this ifdef.

File: content_handler/runtime_holder.cc [hunk 3]
```diff
@@ -448,9 +454,31 @@ void RuntimeHolder::InitMozartInternal() {
...
+    return;
+#else
+    fxl::UniqueFD root_dir(fdio_ns_opendir(namespc_));
+    if (!root_dir.is_valid()) {
+      FXL_LOG(ERROR) << "Unable to load root dir";
+      return;
+    }
+    fxl::UniqueFD data_dir(openat(root_dir.get(),
+                                  kFuchsiaPackageResourceDirectory,
+				  O_RDONLY)));
```
Comment (abarth) [inline content_handler/runtime_holder.cc]: Bad indent.  Also, you should pass O_DIRECTORY

**Comment by abarth** [comment]
LGTM

**Pull request merged**